### PR TITLE
Data scienceemotion prediction with semi supervised learning using the rc algorithm

### LINKED ISSUE #6896: Data Science: Emotion Prediction with Semi-Supervised Learning using the RC Algorithm
*NOTE*: All **1st-time contributors** should know Topic approval and PR submission does NOT guarantee your Topic/Article will be published. Our team of Peer Reviewers and Content Moderators will review all PRs that come in to make sure they adhere to the standard of quality we expect from the EngEd community. 

We expect all community memebers to go through provided examples, resources, and previously published material before submitting content. As a rule of thumb, please only submit articles (pull requests) that are complete, formatted correctly, and include a fully polished article (ready to be published and error free). 

All revisions and edits should be *completed on your own forked repo (as to not take up room in the queue)*. **Any PR that is submitted incomplete (meaning not ready to be published as is) will be closed.** A PR being closed does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.

**NOTE: Any topic suggestion copied/taken/modified from YouTube may Lead to a permanent ban from the community.**

### Topic Suggestion
Enter topic suggestion here...

#### Pre-submission advice
By following all our pre-submission advice and reviewing our Resources folder, you will maximise your chances of your topic being approved.

We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner.

Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published (subject to vary depending on _volume and/or backlog of articles_).

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles that will help you propose and write a successful article.

**Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Writing sample(s): 
Include any links or writing samples - to help our team better gauge your writing quality.

#### Proposed title of article
Data Science: Emotion Prediction with Semi-Supervised Learning with RC Algorithm in Machine Learning

Use keyword research to improve your article's SEO.

#### Proposed article introduction
We learn new skills by observing and practicing, but we also remember, on a deeper level, things we have experienced. This is what the brain does, it creates a model of the world that allows us to respond to new situations in our lives. We often use those feelings and models to predict what other people are feeling by observing their facial expressions, body language, or even their tone of voice. Data science, when applied to computer algorithms, can help us to predict what other people are thinking or feeling by analyzing data. This post will be about learning about emotion prediction with semi-supervised learning with the RC algorithm in machine learning.

For example, if the article is based on machine learning, use the following: [Machine learning] Introduction to Machine Learning.
If the article is based on developing an Android application, use the following: [Android] Developing Apps using Android.

ONLY viable topics to pick from:
- API
- Artificial Intelligence
- Blockchain
- Containers
- Edge Computing
- Languages
- Machine Learning
- Networking
- Node.js
- Security

#### Key takeaways 
-  What is the RC algorithm?
- The problem with current methods
- The RC Algorithm
- Emotion prediction with semi-supervised learning with the RC algorithm in machine learning.
#### Article quality
In this article, we will explore the use of a semi-supervised learning algorithm called the Random Forest (RF) with the Reducing Covariance (RC) technique for emotion prediction. The RC algorithm is used to reduce the dimensionality of the data, making it more suitable for training the RF model. The RF model is then used to predict the emotions of a test data set. The results show that the RC-RF algorithm outperforms other semi-supervised learning algorithms for emotion prediction, with an accuracy of 85.71%.
#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by crezyj**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6896

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by louisefindlay23** [comment]
Plagiarism check complete

File: content/articles/data-science:emotion-prediction-with-semi-supervised-learning-using-the-RC-algorithm/index.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+"The precision is how well the model predicts the labels of the data points it has seen before." 
+"F1 score is a weighted average of two metrics: precision and recall (inverse document frequency)." 
+"L2-norm is the square root of the sum of squares. It is used to normalize the sum of squared errors, which are the difference between the model's output (e.g., predicted class label) and its training example."
+"Euclidean distance is a common measure of similarity in data. Euclidean distance is based on two main parameters: a set of vectors and an axis perpendicular to that set. Given data points (x1, x2...xn), we can consider the set of all pairs of them (x1, x2,...xn). The Euclidean distance between them (d(x1, x2)) is the length of the line segment connecting them. The cosine similarity is a common shorthand version: 1 - d(x1, x2)/d(x1, x3) * d(x2, x3)."
```
Comment (jerimkaura) [inline content/articles/data-science:emotion-prediction-with-semi-supervised-learning-using-the-RC-algorithm/index.md:65]: This section is improperly formatted. Ensure it is clear and precise. If you are listing points, use a bulleted list.

File: content/articles/data-science:emotion-prediction-with-semi-supervised-learning-using-the-RC-algorithm/index.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
+### Emotion Prediction with Semi-Supervised Learning with RC Algorithm in Machine Learning
+Machine learning is the branch of artificial intelligence that uses programming models, statistical techniques, and systems capable of learning from data to improve performance and make decisions automatically. This article will learn how to predict emotions using RC Algorithm in Machine Learning. We will write some code on the same to illustrate this.
+### Table of Content 
+Data set 
+Laoding data
+Model Training
+Model Building
```
Comment (jerimkaura) [inline content/articles/data-science:emotion-prediction-with-semi-supervised-learning-using-the-RC-algorithm/index.md:7]: Check docs for proper formatting of the table of contents.

**Review by jerimkaura — CHANGES_REQUESTED** [review]
Hello @crezyj 

Welcome to the section and thank you for submitting your first article for review.

However, please go over the docs and published articles to understand the procedure well. Format your article for readability and ensure your code snippets are formatted as well.\

Separate your code snippets from the explanation to ensure clarity.

File: content/articles/data-science:emotion-prediction-with-semi-supervised-learning-using-the-RC-algorithm/index.md (hunk 1 above)
Comment (crezyj) [reply content/articles/data-science:emotion-prediction-with-semi-supervised-learning-using-the-RC-algorithm/index.md:65]: noted

File: content/articles/data-science:emotion-prediction-with-semi-supervised-learning-using-the-RC-algorithm/index.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+### Prerequisites 
+Understand how to code Python 3.
+Understand machine learning dependecies addition.
+### Data Set
+To start, let's use Kaggle dataset which has emotion labels for a set of pictures 
+Lets import the important dependencies: like numpy, seaborn, matplotlib, tensorflow.
+```python
```
[the remark this answers is not in the document]
Comment (crezyj) [reply content/articles/data-science:emotion-prediction-with-semi-supervised-learning-using-the-RC-algorithm/index.md:14]: noted

**Comment by hectorkambow** [comment]
This article seems to be a bit light on substance & the PR seems far from being ready to be published. (so far)

We will be closing this PR to make room in the queue - review the docs to get a better idea on style and quality.

**Pull request closed without merging**